### PR TITLE
fixed bandwidth calculation when idle

### PR DESCRIPTION
**Pull request by mbfoss**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -278,6 +278,8 @@ int main (int argc, char** argv)
 				std::cerr << "Error dispatching: " << retval << std::endl;
 			else if (retval != 0)
 				packets_read = true;
+			else
+				gettimeofday(&curtime, NULL);
```
Comment (raboof) [inline main.cpp]: a few lines down we're also doing `time_t const now = ::time(NULL)`. Perhaps we should combine the two?

Comment (mb-gh) [inline main.cpp]: done

File: main.cpp [hunk 2]
```diff
@@ -278,12 +278,13 @@ int main (int argc, char** argv)
 				std::cerr << "Error dispatching: " << retval << std::endl;
 			else if (retval != 0)
 				packets_read = true;
+			else
```
Comment (raboof) [inline main.cpp]: Do we want this 'else'? Perhaps we should update `curtime` regardsless of whether there were any packets read?

Comment (mb-gh) [inline main.cpp]: There is a small problem with that.
When packets are received, `curtime` is updated inside the callbakc called by `pcap_dispath`
The bandwith calculation is done inside `do_refresh()` but it uses

Comment (mb-gh) [inline main.cpp]: So I am afraid there would be some bug due to the the following code

	int process_tcp (u_char * userdata, const dp_header * header, const u_char * m_packet) {
	struct dpargs * args = (struct dpargs *) userdata;
	struct tcphdr * tcp = (struct tcphdr *) m_packet;

	curtime = header->ts;

Comment (mb-gh) [inline main.cpp]: We could delete the assignment of curtime in `process_tcp` and `process_udp` and remove the `else` but it needs to be tested to ensure the bandwith calculation does not become wrong

Comment (raboof) [inline main.cpp]: Ah, that is indeed non-obvious. So for cleaning up old connections we use the time from the packets, where for refreshing the page we use the time from `gettimeofday`.

That seems like a good reason to keep the two separate, can we do that?

Comment (mb-gh) [inline main.cpp]: I think so

Comment (mb-gh) [inline main.cpp]: Still it's not a very clean solution I think.
May be in the future, a better redesign of the code will fix this kind of logic.

**Pull request closed without merging**